server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/variables/ElementaryNodes.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2021 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+
+package org.eclipse.lsp.cobol.core.model.tree.variables;
+
+import com.google.common.collect.ImmutableList;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.lsp.cobol.core.messages.MessageTemplate;
+import org.eclipse.lsp.cobol.core.model.ErrorSeverity;
+import org.eclipse.lsp.cobol.core.model.Locality;
+import org.eclipse.lsp.cobol.core.model.ResultWithErrors;
+import org.eclipse.lsp.cobol.core.model.SyntaxError;
+import org.eclipse.lsp.cobol.core.model.variables.UsageFormat;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+
+import static org.eclipse.lsp.cobol.core.model.tree.variables.EffectiveDataType.UNDETERMINED;
+import static org.eclipse.lsp.cobol.core.model.tree.variables.VariableDefinitionUtil.FALSE_RESULT_WITH_NO_ERRORS;
+
+/**
+ * Abstract class for all elementary variables. These variables actually has space allocations and
+ * are not logical.
+ */
+@Getter
+@ToString(callSuper = true)
+@EqualsAndHashCode(callSuper = true)
+abstract class ElementaryNodes extends VariableWithLevelNode implements IUsageClause {
```

Comment:
I don't mind the class name, but it sounds like a name for an enum

Reply:
Changed it to ElementaryNode, can't think of something else. Any suggestion?

Reply:
Why not, it works for me.